pyccel/ast/numpyext.py

```diff
@@ -1185,11 +1186,11 @@ class NumpyArctanh(NumpyUfuncUnary) : pass
 
 #=======================================================================================
 
-class NumpyAbs(NumpyUfuncUnary):
+class NumpyAbsolute(NumpyUfuncUnary):
     def __init__(self, x):
         self._shape     = x.shape
         self._rank      = x.rank
-        self._dtype     = NativeInteger() if x.dtype is NativeInteger() else NativeReal()
+        self._dtype     = x.dtype
```

Comment:
there are no tests for complex numbers.
I think the support of complex numbers will be in a separate PR for both Fortran and C printers.
what do you think?

Reply:
I think the lack of tests for complexes is an oversight on my part. I think I copied the first half of the tests over from `test_math_funcs.py`. I think they are already supported but just not tested.
I am happy to leave the tests for another PR as there will probably be lots to add again, but I would revert this line which would break any existing support (I think this has already been done)